src/ejabberd_auth_pki.erl

```diff
@@ -0,0 +1,117 @@
+-module(ejabberd_auth_pki).
```

Comment:
doc header.
license header.

Reply:
i would add it with a big pleasure, do we have some doc that describes the format of license/doc header?

Reply:
Be like @kzemek 

```erlang
%%==============================================================================     
%% Copyright 2017 Erlang Solutions Ltd.                                              
%%                                                                                   
%% Licensed under the Apache License, Version 2.0 (the "License");                   
%% you may not use this file except in compliance with the License.                  
%% You may obtain a copy of the License at                                           
%%                                                                                   
%% http://www.apache.org/licenses/LICENSE-2.0                                        
%%                                                                                   
%% Unless required by applicable law or agreed to in writing, software               
%% distributed under the License is distributed on an "AS IS" BASIS,                 
%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.          
%% See the License for the specific language governing permissions and               
%% limitations under the License.                                                    
%%==============================================================================     
                                                                                     
-module(mod_global_distrib_utils).                                                   
-author('konrad.zemek@erlang-solutions.com').                                        
                                                                              
                      
```

It's for *new* files, not based on ejabberd's code.

Reply:
It's in a *private company* dev-handbook.
https://github.com/esl/dev-handbook/blob/master/development_handbook.markdown#module-header

We *should* have subset of it for public use, but we **don't** have yet.

Reply:
@arcusfelis 

I'm not so sure if it's OK to put Apache license in our new modules, because ejabberd files already have GPL license. I'm not that good with legal stuff but isn't there a chance of some conflict here?

Standard company header (according to the devbook) is an optimal choice I guess.

Reply:
@fenek We license the whole project under terms covered in [COPYING file](https://github.com/esl/MongooseIM/blob/master/COPYING). I would interpret files with Apache license header to be *additionally* available under Apache license. Anyway, correct copyright headers helps us see which files still prevent us from moving the whole project from GPL to Apache.

Reply:
It sounds sane but as far as I understand, we are operating on our assumptions, correct? Not a formal opinion of a lawyer experienced in open source licenses.

Anyway, the truth is that the standard header for dev handbook provides more useful information than the one with Apache license... they can be combined so a header fills the first screen of a source code view. :D

OK, now seriously: I'd prefer the handbook header. And this legal stuff is a mess.